pkg/cpio/utils.go

```diff
@@ -256,3 +256,15 @@ func Equal(r Record, s Record) bool {
 	}
 	return uio.ReaderAtEqual(r.ReaderAt, s.ReaderAt)
 }
+
+func ZipSlip(dest, archiveName string) error {
+	clean := filepath.Clean(archiveName)
+	relDest, err := filepath.Rel(dest, clean)
+	if err != nil {
+		return fmt.Errorf("ZipSlip test can not form a path from %q, %q", dest, clean)
+	}
+	if strings.HasPrefix(relDest, "..") {
+		return fmt.Errorf("Error: rel path %q has .. as the prefix and hence fails the ZipSlip test", relDest)
```

Comment:
can you explain here why it's being rejected, rather than mentioning ZipSlip? I'm pretty sure users won't know what that is

Reply:
I don't like this anyway. It's much simpler just to do the path creation correctly, or at least I think it is correct, wdyt, to do:
var base, file string
newpath := filepath.Join(base, filepath.Clean(filepath.Join(os.PathSeparator, file)))

When the file has the / put at front via the Join, the Clean swallows up all the .. in the file name, such that when joined to the base, ZipSlip can't happen. Then all this silliness is not needed.